airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/StatePersistence.java

```diff
@@ -107,10 +108,8 @@ public void updateOrCreateState(final UUID connectionId, final StateWrapper stat
     });
   }
 
-  public Boolean isMigration(final UUID connectionId, final StateType currentStateType, final Optional<StateWrapper> previousState)
-      throws IOException {
-    return previousState.isPresent() && previousState.get().getStateType() == StateType.LEGACY &&
-        currentStateType != StateType.LEGACY;
+  private Boolean isMigration(final StateType currentStateType, final Optional<StateWrapper> previousState) {
+    return previousState.isPresent() && StateMessageHelper.isMigration(currentStateType, previousState.get().getStateType());
```

Comment:
@gosusnp discussion from parent PR:
![image](https://user-images.githubusercontent.com/7798112/186285648-a545a1b9-46df-4243-a995-10ca935d795d.png)

Reply:
@gosusnp 

I simply moved this function into `StateMessageHelper` and added that as an overloaded function - because the original `isMigration` static function in the helper was used here as well: https://github.com/airbytehq/airbyte/blob/master/airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/PersistStateActivityImpl.java#L39

Reply:
Absolutely, it is something we can address later on. Should not be part of this change.